napari/_qt/dialogs/qt_plugin_dialog.py

```diff
@@ -290,10 +334,16 @@ def tag_outdated(self, project_info: ProjectInfo):
 
 
 class QtPluginDialog(QDialog):
-    def __init__(self, parent=None):
+    state_changed = Signal(dict)
+    on_changed = Signal()  # emitted when call order changes
```

Comment:
This feels like a potential dilution of the "single source of truth".  Part of the reason that this dialog didn't need to have any signals before, is that nobody else is was "subscribing" to it (it's just a "view" onto the state of the plugin manager ... which is the _real_ source of truth here).

Currently, in this PR ... we have `QtPluginDialog` emitting a `state_changed` signal that the `QMainWindow` is listening for... and then the `QMainWindow._update_plugins` handles updating the `plugin_manager`.  Why involve QMainwindow at all here?  Why not just have the PluginDialog itself do it (it's primary purpose is to interact with the plugin manager anyway).  Similarly the `on_changed` -> `_update_call_order` signal seems to be involving the QMainWindow in something it doesn't need to be involved in.

In general, the less logic we can put on the main window, and the more we can put into temporary dialogs that pop up, do their business, and go away without needing to create connections across multiple objects, the better.

Reply:
> In general, the less logic we can put on the main window, and the more we can put into temporary dialogs that pop up, do their business, and go away without needing to create connections across multiple objects, the better.

I would say the less logic we put on temporary dialogs the better, as you said, they are just viewers and should be as dumb as possible.

Reply:
Something has to modify the model (the plugin manager)... and I don't see why that should be the main window?

Reply:
Cause the main window is the closest to a "controller" in the MVC paradigm.

Reply:
strongly disagree on this one.  First, we don't _have_ an MVC (we have a MV, like most [Qt patterns](https://doc.qt.io/qt-5/model-view-programming.html)).  Second, while the general concept of a "main window" might be a reasonable object to act as a controller... that's just not how napari is setup.  our main window is _very_ dumb.  Currently, you can open up a QtPluginDialog without a main window at all, interact with the general state of the plugins, and you'll be fine.  No state becomes inconsistent.  Easy to test in isolation. etc...  
With this change, you _must_ have a QMainWindow to test or use any of it.  There's really nothing else on the main window that makes it a good candidate for this.  Looking at the `QMainWindow._update_plugins` method and you'll see it basically doesn't use the `self` object at all.  It's just a plain function that _happens_ to live on the QMainWindow...

Reply:
Ok 👍🏼

Reply:
Hi @tlambert03, thanks again for your feedback and explanations.   I moved all the changes to the plugin manager inside the plugin dialog and off the main window.  This includes both changes to state and call order.  Please have a look and let me know what you think.